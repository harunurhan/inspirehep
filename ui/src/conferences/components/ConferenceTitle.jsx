import React from 'react';
import PropTypes from 'prop-types';
import { Map } from 'immutable';

function ConferenceTitle({ title, acronym }) {
  const mainTitle = title.get('title');
  const subTitle = title.get('subtitle');
  return (
    <span>
      {mainTitle}
      {subTitle && (
        <span>
          <span> : </span>
          {subTitle}
        </span>
      )}
      {acronym && <span> ({acronym})</span>}
    </span>
  );
}

ConferenceTitle.propTypes = {
  acronym: PropTypes.string,
  title: PropTypes.instanceOf(Map).isRequired,
};

export default ConferenceTitle;
