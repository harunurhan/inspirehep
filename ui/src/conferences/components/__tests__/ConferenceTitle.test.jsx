import React from 'react';
import { shallow } from 'enzyme';
import { fromJS } from 'immutable';

import ConferenceTitle from '../ConferenceTitle';

describe('ConferenceTitle', () => {
  it('renders with only title', () => {
    const title = fromJS({ title: 'Conference Title' });
    const wrapper = shallow(<ConferenceTitle title={title} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with also subtitle', () => {
    const title = fromJS({ title: 'Conference Title', subtitle: 'Sub' });
    const wrapper = shallow(<ConferenceTitle title={title} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders with everything', () => {
    const title = fromJS({ title: 'Conference Title', subtitle: 'Sub' });
    const acronym = 'CTest';
    const wrapper = shallow(
      <ConferenceTitle title={title} acronym={acronym} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
