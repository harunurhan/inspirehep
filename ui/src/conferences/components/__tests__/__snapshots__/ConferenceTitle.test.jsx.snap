// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConferenceTitle renders with also subtitle 1`] = `
<span>
  Conference Title
  <span>
    <span>
       : 
    </span>
    Sub
  </span>
</span>
`;

exports[`ConferenceTitle renders with everything 1`] = `
<span>
  Conference Title
  <span>
    <span>
       : 
    </span>
    Sub
  </span>
  <span>
     (
    CTest
    )
  </span>
</span>
`;

exports[`ConferenceTitle renders with only title 1`] = `
<span>
  Conference Title
</span>
`;
